### ISSUE TITLE
VR SecureGo plus

### ISSUE THREAD

**Issue by Dark-Aeon** [PrivSec-dev/banking-apps-compat-report#477]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

VR SecureGo plus

### Link to app

https://play.google.com/store/apps/details?id=de.fiduciagad.securego.vr

### App version

3.6.3

### Country of the app

Germany

### Build Number

2024071600

### Device list

Pixel 7

### Profile app tested in

Owner profile

### Google Play installed?

Installed

### Where did you install this app from?

Google Play Store

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### Native code debugging

- [X] Allowed
- [ ] Blocked

### Exploit protection compatibility mode

- [ ] Enabled
- [X] Disabled

### Memory tagging extension (MTE)

- [ ] Enabled
- [ ] Disabled
- [X] N/A

### Stock OS compatibility

- [ ] Works
- [ ] Does not work
- [X] Not tested

### NFC payments

- [ ] Works
- [ ] Works but requires another service
- [ ] Does not work
- [X] N/A (Not supported by app)
- [ ] Not tested

### Description of the app's functionality

The App is used as MFA to log into online banking and to confirm transactions of Volksbanken and Reifeisenbanken  in Germany

App installed and paired with my bank account no problem and  confirming the login into online banking worked. have not tried confirming a transaction yet.

### Are there any extra notes you think users should know about?

The App is made by the same company that makes the SecureGo Plus App. (Already on the list: https://github.com/PrivSec-dev/banking-apps-compat-report/issues/161)

Might be the same app just with different branding from the bank.

**Comment by spring-onion** [comment]
Cool cool, adding!

**Comment by Dark-Aeon** [comment]
made a wire transaction and confirming with the VR SecureGo plus app worked.

**Comment by kuflierl** [comment]
> I want to add, that this app doesn't require Google Play Services and it can be installed with Aurora.

Yes but no. You can install it usung Aurora but the app still requires you to have Google play services installed though you do not have to be logged in. 


> [@Dark-Aeon](https://github.com/Dark-Aeon) You haven't tested confirming transactions. I've confirmed logins, transactions and other stuff, it never failed.

Can agree, i have completed countless transactions

**Comment by kuflierl** [comment]
> The app stopped working for me since the update to version 4.2.0. It just closes and redirects to https://atruvia.de/sicherheitshinweis

Thank you for letting us know. I will refrain from updating until I have a bypass and or an alternate tan generator.

I hope this is not a safetynet type situation.

**Comment by Dark-Aeon** [comment]
I have the App on version 4.2.0 and I just tried confirming a money transfer with it and it worked no problem.

I have installed the app with google playstore.

Since my initial report I never had any problems.

**Comment by Dark-Aeon** [comment]
> I have the App on version 4.2.0 and I just tried confirming a money transfer with it and it worked no problem.
> 
> I have installed the app with google playstore.
> 
> Since my initial report I never had any problems.

Checked the App Info:

- Exploid protection compatibility mode: **Disabled**
- Hardened Memory Allocator: **Default (Enabled)**
- native code debugging: **Default (Allowed)**
- WebBiew JIT: **Default(Allowed)**
- Dynamic code loading via memory: **Default(Allowed)**
- Dynamic code loading via storage: **Default(Allowed)**

I am still using a Pixel 7, so I have no MTE

**Comment by kuflierl** [comment]
I just updated to version 4.2.0 and everything seems to work just fine.

My permissions: 

Exploid protection compatibility mode: Disabled
Hardened Memory Allocator: Default (Enabled)
Memory Tagging: Default (Enabled) 
Native Code Debugging: Allowed
WebView JIT: Default(Allowed)
Dynamic code loading via memory: Allowed
Dynamic code loading via storage: Default(Allowed)

I have Google play services running in the same (secondary) profile. Also app is installed from the play store using a burner account.

Some apps check if they were really installed from the Google play store and not aurora (eg TK Krankenkasse). A way to forcefully reinstall from the playstore is to "archive" the app and manually install it from the play store.

**Comment by innir** [comment]
Works again for me since version 4.2.3 :-)

**Comment by varia-fur** [comment]
Installed normal, as instructed 
Both without and with secure app spawning
 launches the webpage and closes,
 tried changing the comparability settings multiple times, yet doesn't seem to get out of it 
App Version is 4.2.7
On latest graphene on pixel 8

**Comment by Dark-Aeon** [comment]
update:
I switched to a Pixel 9 now and the App works.  Installed Google Services and then installed the App with the Google Playstore. followed the Phone change instructions of the bank and it worked.

**Comment by rookie-ger** [comment]
Could it be that in its latest version 4.3.1 the app doesn't use the Google Play Integrity API anymore?
The notifications that the App uses this API don't show up anymore.

**Comment by kuflierl** [comment]
> Could it be that in its latest version 4.3.1 the app doesn't use the Google Play Integrity API anymore? The notifications that the App uses this API don't show up anymore.

Can reproduce. The app has also gotten significantly more responsive

**Comment by 80G5q1** [comment]
Tested Version 4.3.1 on my Pixel 9 successfully with WERO. Used the app integrated QR scanner for that

(with activated WERO in my banking app)

**Comment by WebSchw** [comment]
Tested Version 4.3.1 on my Pixel 10 Pro XL in a separate profile installed from Aurora. Google play is installed. Play services has no network, SecureGo had DCL memory/storage and Native code debugging enabled as requested by the app.

Installation works fine, signing off transaction as well.

Switching to another device (Settings -> Device change -> Prepare for device change) can cause problems - the app might silently crash upon selecting this option. Reading out the crash report reveals "[...] getBluetoothLeScanner must not be null [...]". Enabling Bluetooth manually beforehand solves the issue.
It appears that the app only asks to "turn on Bluetooth" once after asking for the "Nearby devices" permission - never again after that. So if you revoke this permission and try to select the device change option again, it will ask for both the permission and Bluetooth activation and it should work.

This issue should affect most of us (given we want to switch to another device) since most of us activated the app by moving from another phone to GOS during which the "Nearby devices" permission was requested.

Furthermore: For switching devices the app needs Location permission and location turned on which it of course does NOT request - on both GOS and stock Android (my receiver in this case).
